# Análisis Exploratorio del Titanic

**Curso:** Programación II — Ciencia de Datos  
**Universidad:** Externado de Colombia  
**Integrantes:** Jacobo Beltrán y Helen Rueda

## Introducción

El dataset del Titanic contiene información sobre los pasajeros del famoso
barco que naufragó en 1912. El objetivo de este análisis es explorar los
datos, entender las variables y responder preguntas sobre los pasajeros
y sus probabilidades de supervivencia.

Los datos provienen de Kaggle y están divididos en dos conjuntos:
- **train.csv**: 891 pasajeros con información de supervivencia
- **test.csv**: 418 pasajeros sin información de supervivencia

## 1. Importación de librerías y carga de datos

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

plt.style.use('seaborn-v0_8')
sns.set_palette('husl')

print("Librerias importadas correctamente")
print(f"Version de Pandas: {pd.__version__}")
print(f"Version de NumPy: {np.__version__}")

Librerias importadas correctamente
Version de Pandas: 3.0.3
Version de NumPy: 2.4.6


In [20]:
import os

base = os.path.dirname(os.path.abspath('proyecto_titanic.ipynb'))
ruta_train = os.path.join(base, '..', 'data', 'train.csv')
ruta_test = os.path.join(base, '..', 'data', 'test.csv')

df_train = pd.read_csv(ruta_train)
df_test = pd.read_csv(ruta_test)

print("Datasets cargados correctamente")
print(f"Filas en train: {df_train.shape[0]} | Columnas en train: {df_train.shape[1]}")
print(f"Filas en test:  {df_test.shape[0]} | Columnas en test:  {df_test.shape[1]}")

Datasets cargados correctamente
Filas en train: 891 | Columnas en train: 12
Filas en test:  418 | Columnas en test:  11


In [ ]:
print("=== Primeras filas de train.csv ===")
df_train.head()

=== Primeras filas de train.csv ===


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [ ]:
print("=== Primeras filas de test.csv ===")
df_test.head()

=== Primeras filas de test.csv ===


,PassengerId,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


## 2. Descripción de las variables

El dataset del Titanic contiene las siguientes variables:

| Variable | Tipo | Descripción |
|----------|------|-------------|
| PassengerId | Numérica | Identificador único de cada pasajero |
| Survived | Categórica | Supervivencia (0 = No, 1 = Sí). Solo en train |
| Pclass | Categórica | Clase del tiquete (1 = Primera, 2 = Segunda, 3 = Tercera) |
| Name | Texto | Nombre completo del pasajero |
| Sex | Categórica | Género del pasajero (male/female) |
| Age | Numérica | Edad en años. Puede tener valores faltantes |
| SibSp | Numérica | Número de hermanos o cónyuges a bordo |
| Parch | Numérica | Número de padres o hijos a bordo |
| Ticket | Texto | Número del tiquete |
| Fare | Numérica | Tarifa pagada por el pasajero |
| Cabin | Texto | Número de cabina. Tiene muchos valores faltantes |
| Embarked | Categórica | Puerto de embarque (C = Cherburgo, Q = Queenstown, S = Southampton) |

In [ ]:
print("=== Tipos de datos — train ===")
print(df_train.dtypes)
print(f"\nTotal de variables: {df_train.shape[1]}")

=== Tipos de datos — train ===
PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

Total de variables: 12


In [ ]:
print("=== Tipos de datos — test ===")
print(df_test.dtypes)
print(f"\nTotal de variables: {df_test.shape[1]}")
print("\nNota: test no tiene la variable Survived")

=== Tipos de datos — test ===
PassengerId      int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

Total de variables: 11

Nota: test no tiene la variable Survived


## 3. Estadísticas básicas y comparación entre datasets

In [ ]:
print("=== Estadísticas básicas — train ===")
df_train.describe()

=== Estadísticas básicas — train ===


,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [ ]:
print("=== Estadísticas básicas — test ===")
df_test.describe()

=== Estadísticas básicas — test ===


,PassengerId,Pclass,Age,SibSp,Parch,Fare
count,418.000000,418.000000,332.000000,418.000000,418.000000,417.000000
mean,1100.500000,2.265550,30.272590,0.447368,0.392344,35.627188
std,120.810458,0.841838,14.181209,0.896760,0.981429,55.907576
min,892.000000,1.000000,0.170000,0.000000,0.000000,0.000000
25%,996.250000,1.000000,21.000000,0.000000,0.000000,7.895800
50%,1100.500000,3.000000,27.000000,0.000000,0.000000,14.454200
75%,1204.750000,3.000000,39.000000,1.000000,0.000000,31.500000
max,1309.000000,3.000000,76.000000,8.000000,9.000000,512.329200


In [ ]:
print("=== Comparación entre datasets ===\n")
print(f"{'':20} {'TRAIN':>10} {'TEST':>10}")
print(f"{'Filas':20} {df_train.shape[0]:>10} {df_test.shape[0]:>10}")
print(f"{'Columnas':20} {df_train.shape[1]:>10} {df_test.shape[1]:>10}")
print(f"{'Edad promedio':20} {df_train['Age'].mean():>10.2f} {df_test['Age'].mean():>10.2f}")
print(f"{'Tarifa promedio':20} {df_train['Fare'].mean():>10.2f} {df_test['Fare'].mean():>10.2f}")
print(f"{'% mujeres':20} {(df_train['Sex']=='female').mean()*100:>10.2f} {(df_test['Sex']=='female').mean()*100:>10.2f}")
print(f"\nNota: train tiene la columna Survived, test no la tiene")

=== Comparación entre datasets ===

                          TRAIN       TEST
Filas                       891        418
Columnas                     12         11
Edad promedio             29.70      30.27
Tarifa promedio           32.20      35.63
% mujeres                 35.24      36.36

Nota: train tiene la columna Survived, test no la tiene


### Comparación entre datasets

Al comparar train y test observamos que:
- **train** tiene 891 pasajeros y **test** tiene 418
- La edad promedio y la tarifa promedio son similares entre ambos datasets
- La proporción de mujeres es comparable entre los dos conjuntos
- Esto sugiere que ambos datasets son representativos de la misma población

## 4. Nueva variable: Familiares

In [ ]:
df_train['Familiares'] = df_train['SibSp'] + df_train['Parch']
df_test['Familiares'] = df_test['SibSp'] + df_test['Parch']

print("Variable Familiares creada correctamente")
print(f"\nDistribucion en train:")
print(df_train['Familiares'].value_counts().sort_index())

Variable Familiares creada correctamente

Distribucion en train:
Familiares
0     537
1     161
2     102
3      29
4      15
5      22
6      12
7       6
10      7
Name: count, dtype: int64


### Análisis de la variable Familiares

La variable **Familiares** representa el número total de familiares
que viajaban con cada pasajero a bordo del Titanic.

**Observaciones:**
- La mayoría de los pasajeros viajaban solos (Familiares = 0)
- Los pasajeros con familias pequeñas (1 a 3 familiares) son el
  segundo grupo más grande
- Muy pocos pasajeros viajaban con familias grandes (4 o más)
- Esta variable será útil para analizar si viajar acompañado
  influyó en las probabilidades de supervivencia

## 5. Unión vertical de los datasets

In [9]:
df_train['source'] = 'train'
df_test['source'] = 'test'

df_test['Survived'] = None

print("Columnas en train:", df_train.columns.tolist())
print("Columnas en test: ", df_test.columns.tolist())

NameError: name 'df_train' is not defined

In [10]:
df_total = pd.concat([df_train, df_test], ignore_index=True)

print("Dataset combinado creado correctamente")
print(f"Filas totales: {df_total.shape[0]}")
print(f"Columnas:      {df_total.shape[1]}")
print(f"\nDistribucion por fuente:")
print(df_total['source'].value_counts())

NameError: name 'pd' is not defined

### Decisión sobre la columna Survived en test

El dataset **test** no contiene la variable `Survived` porque
en la competencia de Kaggle es la variable que se debe predecir.

Para poder unir ambos datasets verticalmente sin perder información,
agregamos la columna `Survived` al dataset test con valores `NaN`,
lo que indica que no se conoce la supervivencia de esos pasajeros.

El dataset combinado tiene **1309 pasajeros** en total:
- 891 de train (con información de supervivencia)
- 418 de test (sin información de supervivencia)

## 6. Revision de estadisticas del dataset combinado

In [11]:
# Estadisticas generales del dataset combinado
print("=== Estadisticas del dataset combinado ===")
print(df_total.describe())

print(f"\nForma del dataset: {df_total.shape[0]} filas x {df_total.shape[1]} columnas")
print(f"\nDistribucion por fuente:")
print(df_total['source'].value_counts())

=== Estadisticas del dataset combinado ===


NameError: name 'df_total' is not defined